applications/SwimmingDEMApplication/custom_elements/qsvms_dem_coupled.cpp

```diff
@@ -0,0 +1,859 @@
+                                             const ProcessInfo& rCurrentProcessInfo) {
+        TElementData data;
+        data.Initialize(*this, rCurrentProcessInfo);
+
+        if (rVariable == ERROR_RATIO)
+        {
+            // Get the element's geometric parameters
+            double Weight = data.Weight;
+            // array_1d<double, TNumNodes> N;
+            // BoundedMatrix<double, TNumNodes, TDim> DN_DX;
+            // GeometryUtils::CalculateGeometryData(this->GetGeometry(), data.DN_DX, data.N, data.Weight);
+            // Calculate this element's fluid properties
+            const double density = this->GetAtCoordinate(data.Density,data.N);
+
+            this->GetEffectiveViscosity(data, data.EffectiveViscosity);
+            const double viscosity = this->GetAtCoordinate(data.EffectiveViscosity,data.N);
+
+            // Get Advective velocity
+            array_1d<double, 3> convective_velocity= this->GetAtCoordinate(data.Velocity, data.N) -
+            this->GetAtCoordinate(data.MeshVelocity, data.N);
+
+            // Output container
+            array_1d< double, 3 > ElementalMomRes(3, 0.0);
+
+            // Calculate stabilization parameter. Note that to estimate the subscale velocity, the dynamic coefficient in TauOne is assumed zero.
+            double tau_one, tau_two;
+            this->CalculateTau(data, convective_velocity, tau_one, tau_two);
+
+            if ( rCurrentProcessInfo[OSS_SWITCH] != 1 ) // ASGS
+            {
+                this->AlgebraicMomentumResidual( data, convective_velocity, ElementalMomRes);
+                ElementalMomRes *= tau_one;
+            }
+            else // OSS
+            {
+                this->OrthogonalMomentumResidual(data, convective_velocity, ElementalMomRes);
+                ElementalMomRes *= tau_one;
+            }
+
+            // Error estimation ( ||U'|| / ||Uh_gauss|| ), taking ||U'|| = TauOne ||MomRes||
+            double ErrorRatio(0.0);//, UNorm(0.0);
+
+            for (unsigned int i = 0; i < Dim; ++i)
+            {
+                ErrorRatio += ElementalMomRes[i] * ElementalMomRes[i];
+            }
+            ErrorRatio = sqrt(ErrorRatio); // / UNorm);
+            ErrorRatio /= density;
+            this->SetValue(ERROR_RATIO, ErrorRatio);
+            rOutput = ErrorRatio;
+        }
+        else if (rVariable == NODAL_AREA)
+        {
+            // Get the element's geometric parameters
+            double Weight = data.Weight;
+            // Carefully write results to nodal variables, to avoid parallelism problems
+            for (unsigned int i = 0; i < NumNodes; ++i)
+            {
+                this->GetGeometry()[i].SetLock(); // So it is safe to write in the node in OpenMP
+                this->GetGeometry()[i].FastGetSolutionStepValue(NODAL_AREA) += Weight * data.N[i];
+                this->GetGeometry()[i].UnSetLock(); // Free the node for other threads
+            }
+        }        
```

Comment:
An `else` statement calling the base element `Calculate` if the provided variable is not implemented is missing.